client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -74,13 +141,37 @@ const CancelPurchaseButton = React.createClass( {
 		} );
 	},
 
+	handleSubmit( error, response ) {
+		if ( error ) {
+			notices.error( error.message );
+			return;
+		}
+
+		notices.success( response.message, { persistent: true } );
+
+		analytics.tracks.recordEvent(
+			'calypso_purchases_cancel_form_submit',
+			{ product_slug: this.props.purchase.productSlug }
+		);
+
+		page.redirect( paths.list() );
+	},
+
+	submitCancelAndRefundPurchase() {
+		this.setState( {
+			submitting: true
+		} );
+
+		cancelAndRefundPurchase( this.props.purchase.id, null, this.handleSubmit );
```

Comment:
We should probably update the other instance that this action is used in `confirm-cancel-purchase/load-endpoint-form.js`